Knowledge_Graph/knowledgegraph.py

```diff
@@ -1,66 +1,130 @@
+import pandas as pd
 import networkx as nx
 import matplotlib.pyplot as plt
-from tokencontainer import GrammarCategories
+import os
 
+from Resources.KnowledgeGraphInfoContainer import KnowledgeGraphInfo
+from Resources.TripleContainer import Triple
+from Resources.JsonWrapper import Content
 
 
-# How the NLP pipeline works:
-# Reciew a text
-# Preprocess the text --> divide text into senteces, remove stopwords etc.
-# Train the model on the text --> output = vectors
-# Send the vectors to the cluster component
-# Create tokens for each word in the text --> Tokens is a class containing different attributes
-# For each word in the text, create a token, insert word + cluster category
+class KnowledgeGraph:
+    column_names = ['subject', 'relation', 'object']
+    dataframe = pd.DataFrame(columns=column_names)
 
-# How the Knowledge Graph works:
-# Evaluate each token, determine whether it is usefull or not
-# By this we mean, if the token is a subject, object etc.. or a stopword, punctuation etc.
-# Determine where in the triple the token should be placed
-# Find a way to save the graph
-# (Print graph at last) - for debugging purpose
+    database_path = ''
 
+    def __init__(self, _database_path):
+        self.database_path = _database_path
 
-class KnowledgeGraph:
-    sentences = []
-    triples = []
+    def generate_triples(self, kg_info: KnowledgeGraphInfo):
+        triples = []
+
+        triples.extend(self.__process_metadata(kg_info.content))
+        triples.extend(self.__analyse_sentences(kg_info.sentences))
+
+        self.__create_branches(triples)
+
+        self.__save_to_file()
```

Comment:
Skal generate tripples save to file? Jeg tænker på om den ikke skal returnere en liste af trippler istedet, også kan gøre noget med dem derfra

Reply:
Fordi den der metode er "metoden udtil" som er den eneste du skal kalde. Man kan lave en ny metode "save_knowledgegraph" som folk så skal kalde efter de har kaldt "generate_tripples" men tænkte ikke det gjorde den store forskel ud over at gøre det mere besværligt for brugeren af klassen

Reply:
Har overvejet at lave en ny klasse som holder ansvaret for at gemme